File: android/app/src/main/kotlin/com/verify/zsy/channels/ZsyMethodChannel.java
package com.verify.zsy.channels;

import android.text.TextUtils;

import org.jetbrains.annotations.NotNull;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;

public class ZsyMethodChannel implements MethodChannel.MethodCallHandler {

    private ZsyActivityHelper zsyActivityHelper;
    private MethodChannel channel;

    private ZsyMethodChannel(ZsyActivityHelper zsyActivityHelper, BinaryMessenger binaryMessenger) {
        this.zsyActivityHelper = zsyActivityHelper;
        channel = new MethodChannel(binaryMessenger, ZsyConstant.METHOD_CHANNEL);
        channel.setMethodCallHandler(this);
    }

    protected static ZsyMethodChannel create(ZsyActivityHelper zsyActivityHelper, BinaryMessenger binaryMessenger) {
        return new ZsyMethodChannel(zsyActivityHelper, binaryMessenger);
    }

    @Override
    public void onMethodCall(@NotNull MethodCall call, @NotNull MethodChannel.Result result) {
        if (TextUtils.equals(call.method, ZsyConstant.BACK_DESKTOP_METHOD)) {
            if (zsyActivityHelper == null) return;
            zsyActivityHelper.getActivity().moveTaskToBack(false);
            result.success(true);
        } else if (TextUtils.equals(call.method, ZsyConstant.QR_SCAN_METHOD)) {
            if (zsyActivityHelper == null) return;
            zsyActivityHelper.startActivityForResult(result, "", ZsyConstant.QR_SCAN_CODE);
            result.success(true);
        } else {
            result.notImplemented();
        }
    }

    protected void onCancel() {
        if (channel != null) {
            channel.setMethodCallHandler(null);
            channel = null;
        }
        if (zsyActivityHelper != null) {
            zsyActivityHelper.onCancel();
            zsyActivityHelper = null;
        }
    }
}

File: android/app/src/main/kotlin/com/verify/zsy/ZsyMethodChannel.kt
package com.verify.zsy

import android.app.Activity
import android.text.TextUtils
import com.verify.zsy.ZsyConstant.BACK_DESKTOP_METHOD
import com.verify.zsy.ZsyConstant.METHOD_CHANNEL
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

/**
 * Android接收Flutter 信息
 */
class ZsyMethodChannel(private val activity: Activity, binaryMessenger: BinaryMessenger) : MethodCallHandler {
    private var methodChannel: MethodChannel? = null
    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        if (TextUtils.equals(call.method, BACK_DESKTOP_METHOD)) {
            activity.moveTaskToBack(false)
            result.success(true)
        }
    }

    init {
        methodChannel = MethodChannel(binaryMessenger, METHOD_CHANNEL)
        methodChannel?.setMethodCallHandler(this)
    }

    fun onCancel() {
        methodChannel?.setMethodCallHandler(null)
        methodChannel = null
    }
}